070-Integrations/024-cloudflare-workers.mdx

```diff
@@ -0,0 +1,146 @@
+
+### Install Xata Client
+
+Inside your project root, run the following:
+
+```sh
+xata init
+```
+
+Accept the defaults. By the end you should have a `.env` file generated for you within your app folder. This file will contain the correct credentials to access your database from your Cloudflare worker.
+
+```bash title=".env"
+XATA_API_KEY=YOUR_API_KEY_HERE
+XATA_BRANCH=main
+```
+
+### Initialize Xata Client
+
+There are two ways of initiailizing Xata client in Cloudflare workers:
+
+1.  ES Modules Syntax (Recommended)
+2.  Service Workers Syntax (Deprecated)
+
+#### ES Modules Syntax (Recommended)
+
+```ts title="src/index.ts"
+import { XataClient } from './xata';
+
+export interface Env {
+  XATA_BRANCH: string;
+  XATA_API_KEY: string;
+}
+
+export default {
+  async fetch(request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
+    const xata = new XataClient({
+      apiKey: env.XATA_API_KEY,
+      branch: env.XATA_BRANCH
+    });
+    // Note that the table name "Posts" may vary
+    // depending on the shape of your schema
+    const posts = await xata.db.Posts.getAll();
+    return new Response(`Total Posts: ${posts.length}`);
+  }
+};
+```
+
+#### Service Workers Syntax (Deprecated)
+
+```ts title="src/index.ts"
+import { XataClient } from './xata';
+
+async function handler(request: Request) {
+  const xata = new XataClient({
+    apiKey: XATA_API_KEY,
+    branch: XATA_BRANCH
+  });
+  // Note that the table name "Posts" may vary
+  // depending on the shape of your schema
+  const posts = await xata.db.Posts.getAll();
+  return new Response(`Total Posts: ${posts.length}`);
+}
+
+// Initialize Worker
+addEventListener('fetch', (event) => {
+  event.respondWith(handler(event.request));
+});
+```
+
+### Local environment variables
+
+Copy your Xata environment values into a new file called .dev.vars at the root of your project. This allows Cloudflare access to them in dev mode.
+
+```bash title=".dev.vars"
+XATA_API_KEY=YOUR_API_KEY_HERE
+XATA_BRANCH=main
+```
+
+### Remote environment variables
+
+In order to make these environment variables accessible in your worker when it is deployed remotely, there are two options:
+
+1.  One click guided workflow from the Cloudflare dashboard (Recommended)
+2.  Manually through the Cloudflare dashboard
+
+#### One click guided workflow (Recommended)
+
+Going through this flow automatically sets the environment variables necessary to interact with your database from a remote Cloudflare worker.
+
+1.  Navigate to the Cloudflare dashboard
+2.  Navigate to your corresponding Worker
+3.  Click Integrations
+4.  Select Xata
+
+![Adding environment variables with our Xata Cloudflare integration](images/cloudflare-integration-env.png)
+
+Your endpoint response should also be available remotely at `<YOUR_WORKER>.<YOUR_SUBDOMAIN>.workers.dev.`
+
+#### Manually adding environment variables
```

Comment:
Cheeky suggestion: 
We could choose not to document this manual flow directly here, and rather link out to Cloudflare's [own docs for this process](https://developers.cloudflare.com/workers/configuration/environment-variables/#add-environment-variables-via-the-dashboard).

This lets us:
1. encourage people to focus on the recommended path
2. reduce the overhead of updating text/screenshots whenever Cloudflare update their docs